### PR TITLE
Downloading the map

### PR DESCRIPTION
**Pull request by emre2038**
## Description
Merging the latest design changes and downloading the current state of the map through a download button

## Related Issue
https://github.com/OpenUpSA/wazimap-ng-ui/issues/104

## How to test it locally
* Run the project
* Click the download button 
* Check if the exported image matches the current map

## Screenshots
Here is a sample exported image

image

### Added
* `map > map_download.js` has the code to export image of the map. Exported image is created using `html2canvas` plugin.

### Updated
* moved `saveAs()` function from `reusable-charts > horizontal-bar-chart.js` to `utils.js` so it can be used from `map > map_download.js` too

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  ⚙️ ran jslint
- [ ]  🧰 ran codeclimate locally

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/map/map_download.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        };
+        let element = document.getElementById("main-map");
+        let self = this;
+
+        html2canvas(element, options).then(function (canvas) {
+            self.saveAs(canvas.toDataURL(), 'map.png');
+        });
+    }
+
+    saveAs = (uri, filename) => {
```
[the remark this answers is not in the document]
Comment (emre2038) [reply src/js/map/map_download.js]: done

File: src/index.html [hunk 2]
```diff
@@ -461,14 +485,21 @@ <h1 class="clear">Map boundary change</h1>
           </div>
         </div>
       </div>
+      <div title="Download map" class="map-download">
```
Comment (milafrerichs) [inline src/index.html:488]: can you remove all other changes and only include the map download button? 
that would be great. 
thanks. 
the other changes should be a separate PR.

File: src/css/webflow.css [hunk 3]
```diff
@@ -1807,6 +1807,7 @@ textarea.w-select {
 .w-condition-invisible {
   display: none !important;
 }
-#dev-tools {
```
Comment (milafrerichs) [inline src/css/webflow.css:1810]: I asked matt to include this. can you leave it in for now? 
thanks.

**Review by milafrerichs — CHANGES_REQUESTED** [review]
can you only add the pieces of HTML and CSS that are needed for this map download change?
that would be great. 
thanks.

File: src/index.html (hunk 2 above)
Comment (emre2038) [reply src/index.html]: done

**Review by milafrerichs — APPROVED** [review]
